### ISSUE TITLE
Failing on Aether

### ISSUE THREAD

**Issue by kenwdelong** [smoope/j2objc-maven-plugin#5]
I have a super-simple project, only a few lines of Java code.

>mvn -version
Apache Maven 3.2.3 (33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4; 2014-08-11T13:58:10-07:00)
Maven home: C:\java\apache-maven-3.2.3\bin\..
Java version: 1.8.0_92, vendor: Oracle Corporation
Java home: c:\java\jdk1.8.0_92\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 7", version: "6.1", arch: "amd64", family: "dos"

`

    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>myco</groupId>
	<artifactId>mylib</artifactId>
	<version>1.0.0-SNAPSHOT</version>

	<dependencies>
		<dependency>
			<groupId>junit</groupId>
			<artifactId>junit</artifactId>
			<version>4.12</version>
			<scope>test</scope>
		</dependency>
	</dependencies>

	<build>
		<plugins>
			<plugin>
				<artifactId>maven-compiler-plugin</artifactId>
				<configuration>
					<source>1.8</source>
					<target>1.8</target>
				</configuration>
			</plugin>

			<plugin>
				<groupId>com.smoope.utils</groupId>
				<artifactId>j2objc-maven-plugin</artifactId>
				<version>1.1.2</version>
				<configuration>
					<j2objcPath>/java/j2objc-1.0.2</j2objcPath>
				</configuration>
			</plugin>
		</plugins>
	</build>

	<scm>
		<connection>scm:git:file://./.git</connection>
	</scm>

</project>`

When I run:

mvn j2objc:convert

I get this stack trace:

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.707 s
[INFO] Finished at: 2016-07-22T11:04:31-07:00
[INFO] Final Memory: 9M/309M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.smoope.utils:j2objc-maven-plugin:1.1.2:convert (default-cli) on project weight-lib: Execution default-cli of goal com.smoope.utils:j2objc-maven-plugin:1.1.2:convert failed: A required class was missing while executing com.smoope.utils:j2objc-maven-plugin:1.1.2:convert: org/eclipse/aether/spi/connector/transport/TransporterFactory
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>com.smoope.utils:j2objc-maven-plugin:1.1.2
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/C:/Users/Ken/.m2/repository/com/smoope/utils/j2objc-maven-plugin/1.1.2/j2objc-maven-plugin-1.1.2.jar
[ERROR] urls[1] = file:/C:/Users/Ken/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[2] = file:/C:/Users/Ken/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[ERROR] urls[3] = file:/C:/Users/Ken/.m2/repository/javax/inject/javax.inject/1/javax.inject-1.jar
[ERROR] urls[4] = file:/C:/Users/Ken/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.2/org.eclipse.sisu.inject-0.3.2.jar
[ERROR] urls[5] = file:/C:/Users/Ken/.m2/repository/org/codehaus/plexus/plexus-utils/1.5.6/plexus-utils-1.5.6.jar
[ERROR] urls[6] = file:/C:/Users/Ken/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.1/plexus-interpolation-1.1.jar
[ERROR] urls[7] = file:/C:/Users/Ken/.m2/repository/org/apache/xbean/xbean-reflect/3.4/xbean-reflect-3.4.jar
[ERROR] urls[8] = file:/C:/Users/Ken/.m2/repository/log4j/log4j/1.2.12/log4j-1.2.12.jar
[ERROR] urls[9] = file:/C:/Users/Ken/.m2/repository/commons-logging/commons-logging-api/1.1/commons-logging-api-1.1.jar
[ERROR] urls[10] = file:/C:/Users/Ken/.m2/repository/com/google/code/google-collections/google-collect/snapshot-20080530/google-collect-snapshot-20080530.jar
[ERROR] urls[11] = file:/C:/Users/Ken/.m2/repository/junit/junit/3.8.2/junit-3.8.2.jar
[ERROR] urls[12] = file:/C:/Users/Ken/.m2/repository/org/codehaus/woodstox/wstx-asl/3.2.6/wstx-asl-3.2.6.jar
[ERROR] urls[13] = file:/C:/Users/Ken/.m2/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar
[ERROR] urls[14] = file:/C:/Users/Ken/.m2/repository/org/sonatype/spice/model-builder/1.3/model-builder-1.3.jar
[ERROR] urls[15] = file:/C:/Users/Ken/.m2/repository/org/apache/maven/maven-project-builder/3.0-alpha-2/maven-project-builder-3.0-alpha-2.jar
[ERROR] urls[16] = file:/C:/Users/Ken/.m2/repository/org/apache/maven/maven-builder-support/3.3.9/maven-builder-support-3.3.9.jar
[ERROR] urls[17] = file:/C:/Users/Ken/.m2/repository/com/google/guava/guava/18.0/guava-18.0.jar
[ERROR] urls[18] = file:/C:/Users/Ken/.m2/repository/org/eclipse/aether/aether-util/1.0.2.v20150114/aether-util-1.0.2.v20150114.jar
[ERROR] urls[19] = file:/C:/Users/Ken/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.6/plexus-component-annotations-1.6.jar
[ERROR] urls[20] = file:/C:/Users/Ken/.m2/repository/org/apache/commons/commons-lang3/3.4/commons-lang3-3.4.jar
[ERROR] urls[21] = file:/C:/Users/Ken/.m2/repository/org/eclipse/aether/aether-connector-basic/1.0.2.v20150114/aether-connector-basic-1.0.2.v20150114.jar
[ERROR] urls[22] = file:/C:/Users/Ken/.m2/repository/org/eclipse/aether/aether-transport-wagon/1.0.2.v20150114/aether-transport-wagon-1.0.2.v20150114.jar
[ERROR] urls[23] = file:/C:/Users/Ken/.m2/repository/org/eclipse/aether/aether-transport-file/1.0.2.v20150114/aether-transport-file-1.0.2.v20150114.jar
[ERROR] urls[24] = file:/C:/Users/Ken/.m2/repository/org/eclipse/aether/aether-transport-http/1.0.2.v20150114/aether-transport-http-1.0.2.v20150114.jar
[ERROR] urls[25] = file:/C:/Users/Ken/.m2/repository/org/apache/httpcomponents/httpclient/4.2.6/httpclient-4.2.6.jar
[ERROR] urls[26] = file:/C:/Users/Ken/.m2/repository/org/apache/httpcomponents/httpcore/4.2.5/httpcore-4.2.5.jar
[ERROR] urls[27] = file:/C:/Users/Ken/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar
[ERROR] urls[28] = file:/C:/Users/Ken/.m2/repository/org/slf4j/jcl-over-slf4j/1.6.2/jcl-over-slf4j-1.6.2.jar
[ERROR] urls[29] = file:/C:/Users/Ken/.m2/repository/org/slf4j/slf4j-api/1.6.2/slf4j-api-1.6.2.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR]
[ERROR] -----------------------------------------------------: org.eclipse.aether.spi.connector.transport.TransporterFactory
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException

When I look in my local Maven Repo, I see the Aether JAR files there, so it seems like they were fetched correctly.  I can't figure out why this won't work!

**Comment by vicmosin** [comment]
Hi,
thank you for your feedback. Could you please share the simple project you tried to convert? So I can try to reproduce the issue. In addition, did you try with a latest maven version?

**Comment by vicmosin** [comment]
Just checked it with 3.2.3 and it's indeed fails. I guess one of the dependencies we use in plugin has some bug..

**Comment by kenwdelong** [comment]
Interestingly, I tried it on Amazon Linux on an EC2 instance and it worked perfectly.  That was with the latest maven, 3.3.9.  

Also, when I upgrade to 3.3.9 on Windows, it worked.  I didn't realize my local Maven was so far out of date.  I ran into other errors, which are probably due to me running Git Bash shell instead of Cygwin, so I'm not going to worry about it.  As long as I can get it to work on my Linux build machine!

Thanks for the super-useful plugin!
